actors/miner/src/lib.rs

```diff
@@ -1950,6 +1958,306 @@ impl Actor {
         Ok(ProveCommitSectors3Return { activation_results: result })
     }
 
+    fn prove_commit_sectors_ni(
+        rt: &impl Runtime,
+        params: ProveCommitSectorsNIParams,
+    ) -> Result<(), ActorError> {
+        let policy = rt.policy();
+        let miner_id = rt.message().receiver().id().unwrap();
+
+        if params.sectors.is_empty() {
+            return Err(actor_error!(illegal_argument, "batch empty"));
+        }
+
+        if !can_prove_commit_ni_seal_proof(rt.policy(), params.seal_proof_type) {
+            return Err(actor_error!(
+                illegal_argument,
+                "unsupported seal proof type {}",
+                i64::from(params.seal_proof_type)
+            ));
+        }
+
+        if params.aggregate_proof.is_empty() {
+            return Err(actor_error!(illegal_argument, "aggregate proof must be non-empty"));
+        }
+
+        if params.aggregate_proof_type != RegisteredAggregateProof::SnarkPackV2 {
+            return Err(actor_error!(illegal_argument, "aggregate proof type must be SnarkPackV2"));
+        }
+
+        let sector_count = params.sectors.len() as u64;
+        if sector_count > policy.max_aggregated_sectors_ni {
+            return Err(actor_error!(
+                illegal_argument,
+                "too many sectors addressed, addressed {} want <= {}",
+                sector_count,
+                policy.max_aggregated_sectors
+            ));
+        }
+        if sector_count < policy.min_aggregated_sectors_ni {
+            return Err(actor_error!(
+                illegal_argument,
+                "too few sectors addressed, addressed {} want >= {}",
+                sector_count,
+                policy.min_aggregated_sectors_ni
+            ));
+        }
```

Comment:
these are done in `validate_seal_aggregate_proof`, so I think we can remove these and also the `is_empty` above